functions/go/gatekeeper-validate/main.go

```diff
@@ -54,11 +54,15 @@ func main() {
 			objects = append(objects, obj)
 		}
 
-		err := Validate(objects)
-		if result, ok := err.(*framework.Result); ok {
+		result, err := Validate(objects)
+		switch {
+		case result != nil && err != nil:
```

Comment:
Why are we using the result if the err != nil? This can sometimes be meaningful (see [Multiple return values](https://golang.org/doc/effective_go#multiple-returns) in Effective Go), but that doesn't appear to be the case here.

Usually you can use a pattern like this:

https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

If you need to handle the result no matter what, I still would avoid the case switches, building up whatever return results gradually.

There also seems to be some inconsistency here as to what is getting returned when, so it's hard for me to say what the best approach would be.